pandas/core/generic.py

```diff
@@ -2647,8 +2657,9 @@ def to_json(
     )
     def to_hdf(
         self,
-        path_or_buf: FilePath | HDFStore,
-        key: str,
+        path: FilePath | HDFStore,
```

Comment:
If someone used to call `to_hdf(path_or_buf=...)` this will now raise because of the missing argument. I think you would need to set a default argument for `path` to avoid this.